featuretools/computational_backends/utils.py

```diff
@@ -291,3 +292,52 @@ def replace_inf_values(feature_matrix, replacement_value=np.nan, columns=None):
     else:
         feature_matrix[columns] = feature_matrix[columns].replace([np.inf, -np.inf], replacement_value)
     return feature_matrix
+
+
+def get_ww_types_from_features(features, entityset, pass_columns=None, cutoff_time=None):
+    '''Given a list of features and entityset (and optionally a list of pass
+    through columns and the cutoff time dataframe), returns the logical types,
+    semantic tags,and origin of each column in the feature matrix.  Both
+    pass_columns and cutoff_time will need to be supplied in order to get the
+    type information for the pass through columns
+    '''
+    if pass_columns is None:
+        pass_columns = []
+    logical_types = {}
+    semantic_tags = {}
+    origins = {}
+
+    for feature in features:
+        names = feature.get_feature_names()
+        for name in names:
+            logical_types[name] = feature.column_schema.logical_type
+            semantic_tags[name] = feature.column_schema.semantic_tags.copy()
+            semantic_tags[name] -= {'index', 'time_index'}
+
+            if logical_types[name] is None and "numeric" in semantic_tags[name]:
+                logical_types[name] = Double
+        if all([f.primitive is None for f in feature.get_dependencies(deep=True)]):
+            origins[name] = "base"
+        else:
+            origins[name] = "engineered"
+
+    for column in pass_columns:
+        logical_types[column] = cutoff_time.ww[column].ww.logical_type
+        semantic_tags[column] = cutoff_time.ww[column].ww.semantic_tags.copy()
+        origins[column] = "base"
+
+    if entityset.dataframe_type in (Library.DASK.value, Library.KOALAS.value):
+        target_dataframe_name = features[0].dataframe_name
+        table_accessor = entityset[target_dataframe_name].ww
```

Comment:
I don't think there's anything wrong with using the accessor like this, so this is just a suggestion for slightly different syntax, but you could store the table schema as a variable, and that might be more in line with how woodwork expects users to access typing info

```python
table_schema = entityset[target_dataframe_name].ww.schema
index_col = table_schema.index
logical_types[index_col] = table_schema.logical_types[index_col]
semantic_tags[index_col] = table_schema.semantic_tags[index_col] # no need to copy, because the .schema already copied

```